### PR TITLE
Updating Blueprint spec to include resource group tags

### PR DESCRIPTION
**Pull request by stuartko**
Updating latest Blueprint preview API to include resource group tagging, which was missing from our existing swagger.

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintDefinition.json [hunk 1]
```diff
@@ -1205,9 +1205,21 @@
             "type": "string",
             "description": "Name of peer artifact."
           }
+        },
+        "tags": {
```
Comment (KrisBash) [inline specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintDefinition.json]: While not a strict rule, this is best done in an incremented api version. If a user with an updated client populates this value, and then a user with a legacy client does a GET/MERGE/PUT to modify the resource, the property would be wiped out.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see my comment

File: specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintDefinition.json (hunk 1 above)
Comment (stuartko) [reply specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintDefinition.json]: This was a risk we were aware of but deemed the risk low enough to not require a version bump. If we weren't in preview we wouldn't take any such risk, even low, but in this case we see it as highly unlikely two users would be modifying the same blueprint with two separate SDK releases. If possible we'd like to push this one through to the existing version.

Comment (KrisBash) [reply specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/blueprintDefinition.json]: Ok

**Review by KrisBash — APPROVED** [review]

**Review by jianghaolu — APPROVED** [review]

**Pull request merged**